### DISCUSSION TITLE
User Authentication

### DISCUSSION THREAD

**Discussion by JoshDevHub** [club-40/odin-pantry#32] [General]
We need to decide on an approach for user authentication with our Rails backend. 

The route we go with *needs* to support JWT for smooth integration with the app's frontend. Ideally we don't store any user data ourselves (so we use a provider of some kind). I can probably be swayed on that if we can't come to a good provider solution but I definitely *don't* want to hand roll this. So the fallback would be some devise/doorkeeper/rodauth thing.

Anyone have any good ideas or tools they know of? I've recently used Auth0 from a different project, and I think their Rails SDK leaves something to be desired. I've heard some good things about [clerk](https://clerk.com/docs/reference/ruby/getting-started) and might check it out. Definitely welcome any thoughts/opinions on this, even from people on the Node side of the curriculum as a lot of auth providers have SDKs in different languages.

**Comment by JaeGif** [comment]
I've used Auth0 with my Node projects before and it worked quite well. It's my goto Node solution, but I don't know anything about the RoR SDK for it. 
Alternatively there is omniauth. It can also be used WITH devise https://github.com/heartcombo/devise#omniauth. 
Here's a quick comparison. 
https://stackshare.io/stackups/auth0-vs-omniauth

I do have to say though, the Clerk documentation is looking real nice. 

Regardless I'm not part of the RoR team, just some suggestions.

**Comment by MarchandMD** [comment]
I recently completed a project using omniauth, omniauth_google_oauth2 and some other gems. 

I just followed a tutorial online. Super easy. Though the Oauth handshake was completed on the front end, and then the response from the Oauth provider was passed to the backend to create a user/session.

**Reply by JoshDevHub** [reply]
This sounds super promising. How interested would you be in helping getting this setup? I'd be willing to pair collab in a call or something on it if you'd like.

"Not interested" is absolutely an acceptable response btw!

**Reply by MarchandMD** [reply]
I'm happy to jump into a call and talk things over and maybe drive/navigate

**Reply by MarchandMD** [reply]
@JoshDevHub did you want to revive this? 

I've made some progress in another project that built enough momentum to start getting open source contributions... so now I'm willing to turn my attention in this direction
